File: src/com/filewatcherapp/models/QueryCriteria.java
package com.filewatcherapp.models;

import java.time.LocalDateTime;

/**
 * Encapsulates the criteria for querying file events from the database.
 */
public class QueryCriteria {
    /** Optional start of date range (inclusive). */
    private LocalDateTime startDate;

    /** Optional end of date range (inclusive). */
    private LocalDateTime endDate;

    /** Optional file extension filter (e.g. ".txt" or "txt"). */
    private String extension;

    /** Optional event type filter: CREATED, DELETED, MODIFIED, RENAMED. */
    private String eventType;

    /** Optional directory or path filter. */
    private String directoryPath;

    public QueryCriteria() {
        // Default constructor
    }

    public LocalDateTime getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDateTime startDate) {
        this.startDate = startDate;
    }

    public LocalDateTime getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDateTime endDate) {
        this.endDate = endDate;
    }

    public String getExtension() {
        return extension;
    }

    public void setExtension(String extension) {
        this.extension = extension;
    }

    public String getEventType() {
        return eventType;
    }

    public void setEventType(String eventType) {
        this.eventType = eventType;
    }

    public String getDirectoryPath() {
        return directoryPath;
    }

    public void setDirectoryPath(String directoryPath) {
        this.directoryPath = directoryPath;
    }

    @Override
    public String toString() {
        return "QueryCriteria{" +
                "startDate=" + startDate +
                ", endDate=" + endDate +
                ", extension='" + extension + '\'' +
                ", eventType='" + eventType + '\'' +
                ", directoryPath='" + directoryPath + '\'' +
                '}';
    }
}

File: FilesystemWatcher/Model/QueryCriteria.cs
namespace FilesystemWatcher.Model
{
    /// <summary>
    /// Represents the criteria for querying file events, including extension, event type,
    /// directory path, and date range.
    /// </summary>
    /// <author>Mansur Yassin</author>
    /// <author>Tairan Zhang</author>
    public class QueryCriteria
    {
        /// <summary>
        /// Gets or sets the file extension to filter by (including the leading dot).
        /// </summary>
        public string Extension { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the type of event to filter by (e.g., "Created", "Changed", "Deleted", "Renamed").
        /// </summary>
        public string EventType { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the directory path in which to search for file events.
        /// </summary>
        public string DirectoryPath { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the start date of the query range.
        /// </summary>
        public DateTime StartDate { get; set; } = DateTime.Today;

        /// <summary>
        /// Gets or sets the end date of the query range.
        /// </summary>
        public DateTime EndDate { get; set; } = DateTime.Today;
    }
}